Automattic/wp-calypso
# Help: Migrate support history to `react-query`

#### Changes proposed in this Pull Request

Currently, the support history (active support tickets) selector is used only in one place. It's the only consumer of the `help.supportHistory` state subtree. That makes this part of the state a good candidate for a `react-query` migration.

This PR creates a new `useActiveSupportTicketsQuery` query with a `withActiveSupportTickets` HoC. It updates `HelpContact` to use that query instead of `<QuerySupportHistory />` and `getActiveSupportTickets()`. It also removes a `<QuerySupportHistory />` usage that was not necessary. Finally, it cleans up the query component and the support history state that is now unused.

Note that at this time, this [query was disabled](https://github.com/Automattic/wp-calypso/blob/cc485bfb786b8655f91d4bdde8d6289e848f0ef5/client/state/help/actions.js#L9) temporarily, so we're carrying that over to the `react-query` implementation (see the `ENABLE_SUPPORT_HISTORY` constant). 

#### Testing instructions

* Checkout the branch locally.
* Go to `/help/contact`
* Create a testing support ticket.
* Change `ENABLE_SUPPORT_HISTORY` to `true`.
* Go to `/help/contact` again.
* Verify that the request to `/support-history` works well, and you can see the notice:

![](https://cldup.com/sa63Vt2mt1.png)
